src/sentry/workflow_engine/handlers/detector/base.py

```diff
@@ -1,17 +1,34 @@
 import abc
 import dataclasses
 import logging
+from collections.abc import Mapping, Sequence
 from typing import Any, Generic, TypeVar
 
-from sentry.issues.issue_occurrence import IssueOccurrence
+from sentry.issues.grouptype import GroupType
+from sentry.issues.issue_occurrence import IssueEvidence, IssueOccurrence
 from sentry.issues.status_change_message import StatusChangeMessage
+from sentry.types.actor import Actor
 from sentry.workflow_engine.models import DataConditionGroup, DataPacket, Detector
 from sentry.workflow_engine.types import DetectorGroupKey, DetectorPriorityLevel
 
 logger = logging.getLogger(__name__)
 T = TypeVar("T")
 
 
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class DetectorOccurrence:
```

Comment:
is there a clean way to link this to the `IssueOccurrence`? i've tried a few approaches, the cleanest is probably to make a shared `BaseOccurrence` class, but this is a less intrusive way to define it.

Reply:
I think a base class is the only way really